helix-term/src/ui/picker.rs

```diff
@@ -118,6 +284,31 @@ impl<T> Picker<T> {
         }
     }
 
+    fn calculate_preview(&mut self, editor: &Editor) {
```

Comment:
This is likely too slow to do synchronously and needs to be done async. You also want to add an upper bound to the cache size because loading hundreds of tree sitter ASTs in memory will quickly consume RAM.

Reply:
As a start it might make sense to completely skip highlighting and simply focus on the preview. Of maybe make it use the `Document` if we find one in `editor.documents`, else just render the file as plaintext?

Reply:
I'll try to make it async and if that doesn't workout in the current form then I guess plaintext rendering is good enough for now.

Reply:
Gave up on async since it needs more work in treesitter parsing.

Reply:
> This is likely too slow to do synchronously and needs to be done async. You also want to add an upper bound to the cache size because loading hundreds of tree sitter ASTs in memory will quickly consume RAM.

I wonder if it would be possible to create a temporary list of `Document`s solely for this purpose, but I think there should be some delay so it doesn't continuously delete/add new `Document`s as you scroll down the picker.